<?xml version="1.0" encoding="UTF-8"?>
<!--
Copyright (C) 2017-2024 Hanson Robotics

This program is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program.  If not, see <https://www.gnu.org/licenses/>.
-->
<launch>
    <arg name="stt"                     default="true" />
    <arg name="speech_recognizer_node"  default="true" />
    <arg name="audio_analyzer_node"     default="true" />
    <arg name="mic_name"                default="unknown_mic" />

    <group ns="hr">
        <group ns="perception">
            <param name="audio_stream_topic" value="/hr/sensors/audio/$(arg mic_name)" />
            <param name="tts_topic" value="/hr/control/speech/say" />
            <param name="sentence_topic" value="hear/sentence" />
            <param name="words_topic" value="hear/words" />
            <param name="chat_event_topic" value="hear/event" />
            <param name="interim_topic" value="hear/interim_speech" />
            <param name="robot_speech_event_topic" value="/hr/control/speech/event" />
            <node name="$(arg mic_name)" pkg="asr" type="recognizer_node.py" respawn="true" if="$(arg speech_recognizer_node)">
                <param name="enable" value="$(arg stt)" />
            </node>
            <node name="audio_analyzer" pkg="asr" type="audio_analyzer.py" respawn="true" if="$(arg audio_analyzer_node)" />
        </group>
    </group>
</launch>

